addons/addon-base-raas/packages/base-raas-services/lib/helpers/utils.js

```diff
@@ -51,8 +51,15 @@ async function updateS3BucketPolicy(s3Client, s3BucketName, s3Policy, revisedSta
       s3Policy.Statement.push(statement);
     }
   });
+  try {
+    await s3Client.putBucketPolicy({ Bucket: s3BucketName, Policy: JSON.stringify(s3Policy) }).promise();
+  } catch (error) {
+    throw this.boom.badRequest(
+      `Error in putting bucket policy: ${JSON.stringify(s3Policy)} into bucket: ${bucketName}`,
```

Comment:
`Failed updating  bucket policy: ${JSON.stringify(s3Policy)} for bucket: ${bucketName}. Check if original bucket policy is too large`